src/routes/goals/handlers.js

```diff
@@ -227,3 +228,40 @@ export async function retrieveGoalByIdAndRecipient(req, res) {
     await handleErrors(req, res, error, `${logContext}:RETRIEVE_GOAL_BY_ID_AND_RECIPIENT`);
   }
 }
+
+export async function getSimilarGoalsForRecipient(req, res) {
+  const recipientId = parseInt(req.params.recipient_id, 10);
+
+  if (Number.isNaN(recipientId)) {
+    return res.status(400).send('Recipient ID must be an integer');
+  }
+
+  const cluster = Object.prototype.hasOwnProperty.call(req.query, 'cluster');
+  const userId = await currentUserId(req, res);
+  const user = await userById(userId);
+
+  try {
+    const similarGoalIds = await similarGoalsForRecipient(recipientId, cluster);
+
+    const ids = Array.from(similarGoalIds.reduce((acc, resp) => {
+      const id1 = parseInt(resp.goal1.id, 10);
+      const id2 = parseInt(resp.goal2.id, 10);
+      acc.add(id1);
+      acc.add(id2);
+      return acc;
+    }, new Set()));
+
+    const canView = await Promise.all(ids.map(async (id) => {
+      const goal = await goalByIdWithActivityReportsAndRegions(id);
+      return new Goal(user, goal).canView();
+    }));
+
+    if (!canView.every((permission) => permission)) {
+      return res.sendStatus(401);
+    }
+
+    return res.json(similarGoalIds);
+  } catch (error) {
+    return res.status(500);
```

Comment:
this should probably use our "handleErrors" function

Reply:
ah, yeah: 6e3e705a5c00e2859eb76d781c362b2aac57d149